src/components/FindMovie/FindMovie.tsx

```diff
@@ -17,41 +68,60 @@ export const FindMovie: React.FC = () => {
               id="movie-title"
               placeholder="Enter a title to search"
               className="input is-dander"
+              value={movieTitleQuery}
+              onChange={(event) => handleFindMovie(event.target.value)}
             />
           </div>
 
-          <p className="help is-danger" data-cy="errorMessage">
-            Can&apos;t find a movie with such a title
-          </p>
+          {hasLoadingError && (
+            <p className="help is-danger" data-cy="errorMessage">
+              Can&apos;t find a movie with such a title
+            </p>
+          )}
+
         </div>
 
         <div className="field is-grouped">
           <div className="control">
+
             <button
+              disabled={!movieTitleQuery}
               data-cy="searchButton"
               type="submit"
-              className="button is-light"
+              className={classNames(
+                'button is-light',
+                { 'is-loading': loadingMovie },
+              )}
+              onClick={handleLoadMovieFromServer}
             >
-              Find a movie
+              {!movie
+                ? 'Find a movie'
+                : 'Search again'}
             </button>
+
           </div>
 
           <div className="control">
-            <button
-              data-cy="addButton"
-              type="button"
-              className="button is-primary"
-            >
-              Add to the list
-            </button>
+            {movie && (
+              <button
+                data-cy="addButton"
+                type="button"
+                className="button is-primary"
+                onClick={handleAddNewMovieToTheList}
```

Comment:
do it on submit

Reply:
I changed it, but I don't know if I correctly understood what I needed to do.
Can you explain, why my solution was wrong and why your solution is better?

Reply:
it wasn't wrong if logic is correct, but will be more semantic and logically  correct to use onSubmit handler if you use form in your solution